src/jit/emitxarch.h

```diff
@@ -28,6 +28,16 @@ inline static bool isDoubleReg(regNumber reg)
 /*         Routines that compute the size of / encode instructions      */
 /************************************************************************/
 
+// code_t is a type used to accumulate bits of opcode + prefixes. On amd64,
+// it must be 64 bits to the REX prefixes. On both x86 and amd64, it must be
+// 64 bits to support AVX, with its 3-byte VEX prefix. For legacy backend
+// (which doesn't support AVX), leave it as size_t.
+#if defined(LEGACY_BACKEND)
+typedef size_t code_t;
+#else  // !defined(LEGACY_BACKEND)
+typedef unsigned __int64 code_t;
+#endif // !defined(LEGACY_BACKEND)
```

Comment:
I thought about it, although currently FEATURE_AVX_SUPPORT == !LEGACY_BACKEND for xarch.